src/client/components/InputSearchMeals.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState } from "react";
+import { useMeals } from "../contexts/MealsContext";
+import { BsSearch } from "react-icons/bs";
+import { FiDelete } from "react-icons/fi";
+import { Link } from "react-router-dom";
+
+function InputSearchMeals() {
+  const { meals } = useMeals();
+  const [filtred, setFiltred] = useState([]);
+  const [value, setValue] = useState("");
+
+  const onChangeInput = (e) => {
+    const searchValue = e.target.value;
+    const filterTitles = meals.filter((meal) => {
+      return meal.title.toLowerCase().includes(searchValue.toLowerCase());
+    });
+    setFiltred(filterTitles);
+    setValue(searchValue);
+  };
+
+  const clearSearch = (e) => {
+    setFiltred([]);
+    setValue("");
+  };
+
+  return (
+    <div className="MealsSearch">
+      <div>
+        <label>
+          <input
+            className="MealsSearchInput"
+            onChange={onChangeInput}
+            placeholder="Search for meals"
+            value={value}
+          />
+          {filtred.length === 0 ? (
```

Comment:
Normally the clear button would appear when the search input is not empty. So it is a bit unexpected to only show the clear button if there were matching results.